framework/selftest.cpp

```diff
@@ -179,6 +180,70 @@ static int selftest_logs_getcpu_run(struct test *test, int cpu)
     return EXIT_SUCCESS;
 }
 
+static int selftest_logs_reschedule_init(struct test *test)
+{
+    // In order to always get the same result and avoid race conditions,
+    // we use semaphores to synchronize the access to reschedule()
+    int sem_size = num_cpus() - 1;
+    sem_t *reschedule_sem = (sem_t *) malloc(sizeof(sem_t)*sem_size);
+    for (int i = 0; i < sem_size; i++) {
+        sem_init(&reschedule_sem[i], 0, 0);
+    }
+
+    test->data = (void *) reschedule_sem;
+
+    return EXIT_SUCCESS;
+}
+
+static int selftest_logs_reschedule_run(struct test *test, int cpu)
+{
+    sem_t *semaphores = (sem_t *) test->data;
+    int cpu_number = -1;
+#if defined(_WIN32)
+    PROCESSOR_NUMBER number;
+    GetCurrentProcessorNumberEx(&number);
+    // see win32/cpu_affinity.cpp
+    cpu_number = number.Group * 64 + number.Number;
+#elif defined(__linux__) || defined(__FreeBSD__)
+    cpu_number = sched_getcpu();
+#else
+    log_skip(OsNotSupportedSkipCategory, "No API to get the CPU number on this OS");
+#endif
```

Comment:
Please deduplicate from the `logs_getcpu` code - just write local `static` function to get the CPU.

For the `#else` case, you can do:
```c
    errno = ENOSYS;
    return -1;
```
and then just leave the "OS failed:" text below.